dmcontent/content_loader.py

```diff
@@ -353,7 +353,24 @@ def get_error_messages(self, errors, question_descriptor_from="label"):
 
         return errors_map
 
+    @staticmethod
+    def unformat_assurance(key, data):
+        return {
+            key + '--assurance': data[key].get('assurance', None),
+            key: data[key].get('value', None)
+        }
+
+    @staticmethod
+    def unformat_date(key, data):
+        result = {}
+        value = data[key]
+        if data[key]:
+            for partial_value, field in zip(value.split('-'), Date.FIELDS):
+                result['-'.join([key, field])] = partial_value
+        return result
+
     def unformat_data(self, data):
+
         """Unpack assurance information to be used in a form
```

Comment:
If we do decide to stick with this solution then documentation for this function needs updating. Also is it worth adding a test to cover `unformat_date` and maybe calling `unformat_data` when passing a date field in?